src/olympia/devhub/templates/devhub/addons/listing/items.html

```diff
@@ -14,7 +14,7 @@
           {{ _("This add-on doesn't have any versions.")}}
         </p>
       {% else %}
-        {% if addon.status != amo.STATUS_NULL %}
+        {% if addon.status != amo.STATUS_NULL and addon.has_listed_versions() %}
```

Comment:
https://github.com/mozilla/addons-server/pull/4494/commits/a358dfb099455042fe8f5abd47a0ce946e871b4c should do it, including fixes to other tests that broke because of the cleanup I did in `test_views.py` regarding how add-ons are created for those tests...